ablySpec/RealtimeClientChannel.swift

```diff
@@ -186,6 +187,56 @@ class RealtimeClientChannel: QuickSpec {
             // RTL6
             describe("publish") {
 
+                // RTL6a
+                it("should encode messages in the same way as the RestChannel") {
+                    let data = ["value":1]
+                    var restEncodedMessage: ARTMessage?
+                    var realtimeEncodedMessage: ARTMessage?
+
+                    let rest = ARTRest(options: AblyTests.commonAppSetup())
+                    let restChannel = rest.channels.get("test")
+
+                    let restBlock: @convention(block) (AspectInfo, ARTMessage) -> Void = { info, message in
+                        let invocation = info.originalInvocation()
+                        invocation.invoke()
```

Comment:
That link shows how `encodeWithEncoder` takes an `output` argument, where it writes the _new_ message, which is then a different instance than the `message` in this block. For this to be correct you need to use the returned message.